functests/1_performance/cpu_management.go

```diff
@@ -349,6 +311,76 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			}
 		})
 	})
+
+	Describe("Verification that IRQ load balance can be disabled per POD", func() {
+		var testpod *corev1.Pod
+
+		BeforeEach(func() {
+			if profile.Spec.GloballyDisableIrqLoadBalancing != nil && *profile.Spec.GloballyDisableIrqLoadBalancing {
+				Skip("IRQ load balance should be enabled (GloballyDisableIrqLoadBalancing=false), skipping test")
+			}
+		})
+
+		AfterEach(func() {
+			deleteTestPod(testpod)
+		})
+
+		It("[test_id:36364] should disable IRQ balance for CPU where POD is running", func() {
+			By("checking default smp affinity is equal to all active CPUs")
+			defaultSmpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(defaultSmpAffinitySet).To(Equal(onlineCPUsSet), fmt.Sprintf("Default SMP Affinity %s should be equal to all active CPUs %s", defaultSmpAffinitySet, onlineCPUsSet))
+
+			By("Running pod with annotations that disable specific CPU from IRQ balancer")
+			annotations := map[string]string{
+				"irq-load-balancing.crio.io": "true",
+				"cpu-quota.crio.io":          "true",
+			}
+			testpod = getTestPodWithAnnotations(annotations)
+
+			err = testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+			runTop := []string{"/bin/bash", "-c", "tmux new -d 'top -b > /dev/null'"}
+			_, err = pods.ExecCommandOnPod(testpod, runTop)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking that the default smp affinity mask was updated and CPU (where 'top' is running) isolated")
+			defaultSmpAffinitySet, err = nodes.GetDefaultSmpAffinitySet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			getPsr := []string{"/bin/bash", "-c", "ps -o psr $(pgrep top) | tail -1"}
+			psr, err := nodes.ExecCommandOnNode(getPsr, workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+			psrSet, err := cpuset.Parse(strings.Trim(psr, " "))
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(defaultSmpAffinitySet).To(Equal(onlineCPUsSet.Difference(psrSet)), fmt.Sprintf("Default SMP affinity should not contain isolated CPU %s", psr))
```

Comment:
Is there any other way to get the cpus assigned to a pod other than running some process inside?
Running  sudo crictl inspectp <podid> will return a json with a "cpu": {  "shares": 2 } construct which seems to be the assigned cpus

Reply:
hm, I think "crictl inspectp" does not show the right cpu, I have 2 pods running and both of them has same "shares":

[root@worker-0 ~]# crictl inspectp 7ed0a6a3b6a4b | grep shares
            "shares": 2
[root@worker-0 ~]# crictl inspectp 1964ca23ccf0e | grep shares
            "shares": 2

I tried some other ways (e.g. parse "sleep 10h" since we anyway have this process running on the test pod) but all of them were a bit more complicated..

Reply:
I wonder if there may be a way to get this info, the cpus the pod is running on, using k8s. @fromanirh @cynepco3hahue do you know anything about it?

Reply:
I would be happy to hear some advice but as I know the pod itself uses full cpu affinity mask and only running processes have the right mask.

Reply:
Are we looking to get the cpu affinity mask of  any running process? If so I'd just use
```
grep Cpus_allowed_list /proc/self/status
```
Or do we need something else?

Reply:
Thanks @fromanirh this is what we were looking for. @dshchedr, instead of running tmux, get its psr while is running and so on, just run the above command in the pod, is cleaner.

Reply:
Nice! thanks @fromanirh !